### PR TITLE
Draft: PPT mode

### PR DESCRIPTION
**Pull request by KSDaemon**
This is a work in progress implementation of PPT Mode in lib for review and discussions

### CODE REVIEW

File: lib/src/message/call.dart [hunk 1]
```diff
@@ -31,5 +32,26 @@ class CallOptions {
...
+      String? ppt_keyid}) {
+      this.ppt_scheme = ppt_scheme;
+      this.ppt_serializer = ppt_serializer;
+      this.ppt_cipher = ppt_cipher;
+      this.ppt_keyid = ppt_keyid;
+  }
+
+  @override
+  bool Verify() {
+      if (timeout! < 0) {
```
Comment (mjohnsullivan) [inline lib/src/message/call.dart:51]: Timeout possibly null at this point?

File: lib/src/message/subscribed.dart [hunk 2]
```diff
@@ -25,7 +25,7 @@ class Subscribed extends AbstractMessage {
     return _revokeCompleter.future;
   }
 
-  /// Is called by the session when the revokation happens
+  /// Is called by the session when the revocation happens
```
Comment (mjohnsullivan) [inline lib/src/message/subscribed.dart:28]: This is what's currently breaking the build. revoke is calling compete with a nullable String, while the completer is a non-nullable string.

**Comment by mjohnsullivan** [comment]
Main comment I have is that most things appear to be marked as nullable and there's liberal use of ! that assumes the data is not null. I haven't looked through the rest of the Connectanum code, so unsure if this is the norm here.

**Comment by KSDaemon** [comment]
Closing as this became obsolete: upstream made some updates and merged, our own fork also went ahead.

**Pull request closed without merging**